src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -150,17 +150,33 @@ void CASEServer::PrepareForSessionEstablishment(const ScopedNodeId & previouslyE
     VerifyOrDie(mPinnedSecureSession.HasValue());
 }
 
+void CASEServer::OnSessionEstablishmentStarted()
+{
+    if (mAppDelegate != nullptr)
+    {
+        mAppDelegate->OnCASEStarted();
```

Comment:
Similarly, we can end up with OnCASEFailed without OnCASEStarted.  Is that the intent?

Reply:
The above discussion holds here as well.